RFC/src/RFC-0301_NamespaceRegistration.md

```diff
@@ -0,0 +1,236 @@
+without notice.
+
+This document may include preliminary concepts that may or may not be in the process of being developed by the Tari
+community. The release of this document is intended solely for review and discussion by the community regarding the
+technological merits of the potential system outlined herein.
+
+## Goals
+
+This purpose of this document is to describe and specify the process for creating and linking an [asset issuer] 
+specified domain name with a [digital asset] on the Digital Assets Network ([DAN]).
+
+
+
+## Related RFCs
+
+- [RFC-0341/AssetRegistration](RFC-0341_AssetRegistration.md)
+- [RFC-0310/AssetTemplates](RFC-0310_AssetTemplates.md)
+
+<br>
+
+# Description
+
+
+
+## Background
+
+In order to easily differentiate different [digital asset]s in the DAN, apart from some unique unpronounceable 
+character string, a human readable identifier (domain name) is required. It is perceived that shorter names will have 
+higher value due to branding and marketability, and the question is how this can be managed elegantly. It is also 
+undesirable if for example the real Disney is forced to use the long versioned "*disney.com-goofy-is-my-asset-yes*" 
+because some fake Disneys claimed "*goofy*" and "*disney.com-goofy*" and everything in between.
+
+One method to curb name space squatting is to register names on the [base layer] layer with a domain name registration 
+transaction. Let us call such a name a Registered Asset Issuer Name (RAIN). To make registering RAINs difficult enough 
+to prevent spamming the network, a certain amount of [Tari coins] must be committed in a burn (permanently destroy) or 
+a time locked pay to self type transaction. Lots of management overhead will be associated with such a scheme, even if 
+domain-less assets are allowed. Although name space squatting will be curbed, it would be impossible to stop someone 
+from registering say a "*disney.com*" RAIN if they do not own the real "*disney.com*" Fully Qualified Domain Name (FQDN).
+
+Another approach would be to make use of the public Domain Name System (DNS) and to link the FQDNs, that are already 
+registered, to the [digital asset]s in the DAN, making use of [OpenAlias](https://openalias.org/) text (TXT) DNS 
+records on a FQDN. Let us call this a Registered Asset Issuer Domain (RAID) TXT record. If we hash a public key and 
+FQDN pair, it will provide us with a unique RAID_ID (RAID Identification). The RAID_ID will serve a similar purpose in 
+the DAN as a Top Level Domain (TLD) in a DNS, as all digital assets belonging to a specific [asset issuer] could then 
+be grouped under the FQDN by inference. To make this scheme more elaborate, but potentially unnecessary, all such 
+RAIN_IDs could be also be registered on the [base layer] layer, similar to the RAIN scheme.
+
+If the standard Mimblewimble protocol is to be followed, a new output feature can be defined to cater for a RAID tuple 
```

Comment:
I dont think we have to create a UTXO. The RAID ID can live outside of the base layer completely. 

We only require validation of the raid_id on asset creation as we only then link the DNS to an asset. So the Asset issuer only needs to supply proof that DAN on asset creation of that. For additional security we could link the Raid ID to the assets so everyone can check that on a later stage. 

For all of this to work we dont require locking down of the RAID_ID anywhere on the base layer. If the user removes the ID from his DNS records it goes invalid. And people other than the asset issuer is not suppose to have access to change his DNS records.

Reply:
Agreed.
This RFC is not suggesting to create a UTXO on the base layer. The text you commented on is only part of a background discussion, which only discussed possibilities. The very last paragraph states: 
"_This document explores the creation and use of RAID TXT records to link asset issuer specified domain names with digital assets on the [DAN], without RAID_IDs being registered on the [base layer]._"

Reply:
The document is hinting at this, but it is never actually stated, or perhaps I mis read it somewhere.

Reply:
See the last paragraph of **Background**